ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/events/[eventName]/logs/[eventId]/EventPayload.tsx

```diff
@@ -29,15 +36,30 @@ export default async function EventPayload({ event }: EventPayloadProps) {
   const formattedPayload = JSON.stringify(parsedPayload, null, 2);
 
   return (
-    <div className="flex h-full flex-col space-y-1.5 rounded-xl bg-slate-900 text-white">
-      <header className="bg-slate-910 flex items-center justify-between rounded-t-xl p-1.5">
-        <h3 className="px-6 py-2.5">Payload</h3>
-      </header>
-      <div className="flex-1 overflow-auto">
-        <div className="p-6">
-          <SyntaxHighlighter language="json">{formattedPayload}</SyntaxHighlighter>
-        </div>
-      </div>
-    </div>
+    <DashboardCodeBlock
+      tabs={[
+        {
+          label: 'Payload',
+          content: formattedPayload,
+          language: 'json',
+          readOnly: true,
+        },
+      ]}
+      actions={
+        isSendToDevServerEnabled
+          ? [
+              {
+                label: 'Send to Dev Server',
+                title: isRunning
+                  ? 'Send event payload to running Dev Server'
+                  : 'Dev Server is not running',
+                icon: <IconLightningBolt />,
+                onClick: () => send(payload),
+                disabled: !isRunning,
```

Comment:
What are your thoughts on polling? Otherwise the user needs to refresh the page after starting the Dev Server

Reply:
Yeah, that's fair. I thought about adding that, but also didn't want to make it too heavy weight. Perhaps we should add it at a reasonable interval. Realistically, it would be cool to make this a setting to disable as well. During the feature flip testing I think it may be ok without polling and see what user feedback that we get.